kubetest/util.go

```diff
@@ -579,18 +579,27 @@ func flushMem() {
 	}
 }
 
+// (only works on gke)
 // getLatestGKEVersion will return newest validMasterVersions
-// only works in gke environment
-func getLatestGKEVersion(project, zone string) (string, error) {
+// pass in releasePrefix to get latest valid version of a specific release
+// empty means use latest across all available releases
+func getLatestGKEVersion(project, zone, releasePrefix string) (string, error) {
 	res, err := output(exec.Command("gcloud", "container", "get-server-config", fmt.Sprintf("--project=%v", project), fmt.Sprintf("--zone=%v", zone), "--format=value(validMasterVersions)"))
 	if err != nil {
 		return "", err
 	}
 	versions := strings.Split(string(res), ";")
-	if len(versions) == 0 {
-		return "", fmt.Errorf("invalid gke master version string: %s", string(res))
+	latestValid := ""
+	for _, version := range versions {
```

Comment:
The assumption is that `versions` is already sorted, is that a safe assumption?

Reply:
yep, we've been relying on this behavior (mentioned offline for internal references).

Reply:
Ack.